mars/dataframe/tests/test_core.py

```diff
@@ -320,3 +320,92 @@ def test_key_value(setup):
 
     result = idx.values.execute().fetch()
     np.testing.assert_array_equal(result, raw.values)
+
+
+def test_between(setup):
+    pd_series = pd.Series(pd.date_range("1/1/2000", periods=10))
+    pd_left, pd_right = pd_series[2], pd_series[7]
+    expected = (pd_series >= pd_left) & (pd_series <= pd_right)
```

Comment:
Also use `between` in pandas code to make sure behaviors between pandas and Mars are consistent.